clang/test/SemaCXX/complete-member-pointers.cpp

```diff
@@ -13,3 +15,18 @@ template <typename T>
 struct S3 {
   int T::*foo;
 };
+
+template<int I> struct Base {};
+struct
+S5 // #S5
+:
+Base<sizeof(int S5::*)>
+// expected-error@-1 {{member pointer has incomplete base type 'S5'}}
```

Comment:
Should this be an error in Microsoft mode? Shouldn't we silently use the unspecified model in that mode?

Reply:
It's an error for `-fcomplete-member-pointers` which is supposed to tell you when you use a member pointer in a way that the MS ABI would have it be unspecified, regardless of if you are compiling with MS member pointers or Itanium member pointers

Reply:
Yes, that's just how `-fcomplete-member-pointers` works currently. It's more like a `-Werror=...` diagnostic